res/controllers/MVave-SMC-Mixer-scripts.js

```diff
@@ -130,22 +130,30 @@ var SMCMixer;
         }
     }
 
-    class EqRack {
+    class EqRack extends components.ComponentContainer {
         constructor(index) {
+            super({});
             const channel = mapIndexToChannel(index);
             this.knob = new Encoder({
                 group: `[Channel${channel}]`,
                 midi: [0xB0, 0x10 + index],
                 inKey: "pregain",
             });
 
+            const _this = this;
             const btnInToggle = () => {
-                const knob = this.knob;
-                const origGroup = this.knob.group;
-                const origInKey = this.knob.inKey;
+                // "this" is undefined here due to a bug in QJSEngine, so we
```

Comment:
`this` also always refers to the EqRack because this is an arrow function. You'll need to declare it as an anonymous function (so `const btnInToggle = function() {`), so the `this` can be appropriately set when this function is getting called.

Reply:
Then `this` should also work correctly (and always refer to the correct button instance instead of the `EqRack` instance).

Reply:
You'd be right except for the bug in QJSEngine I mentioned, if I call super above, this stops being defined entirely and using "this" results in an error. It definitely shouldn't be this way, but it is.

Reply:
Oh, no sorry for the bogus comment. I totally missed where we already return a function in the `btnInToggle` case. Yeah in that case the local alias is probably best.